analyzers/src/SonarAnalyzer.Common/Extensions/RegexExtensions.cs

```diff
@@ -24,6 +24,8 @@ namespace SonarAnalyzer.Extensions;
 
 public static class RegexExtensions
 {
+    private static readonly MatchCollection EmptyMatchCollection = Regex.Matches(string.Empty, "a", RegexOptions.None, RegexConstants.DefaultTimeout);
```

Comment:
`MatchCollection` can not be constructed, so we hack an empty collection ourselves.